src/components/Annotations/BoxPlot.tsx

```diff
@@ -0,0 +1,117 @@
+import { CSSProperties } from 'react';
+
+import { useBoxPlotPosition } from '../../hooks';
+import { ScalarValue } from '../../types';
+
+export interface AnnotationBoxPlotProps {
+  xAxis?: string;
+  yAxis?: string;
+  min: ScalarValue;
+  max: ScalarValue;
+  q1: ScalarValue;
+  median: ScalarValue;
+  q3: ScalarValue;
+  width: ScalarValue;
+  y: ScalarValue;
+  medianWidth?: number | `${number}`;
+  medianColor?: string;
+  medianStyle?: CSSProperties;
+  strokeWidth?: number | `${number}`;
+  strokeColor?: string;
+  boxColor?: string;
+  boxStyle?: CSSProperties;
+  whiskerWidth?: number | `${number}`;
+  whiskerColor?: string;
+  whiskerStyle?: CSSProperties;
+  minMaxWidth?: number | `${number}`;
+  minMaxColor?: string;
+  minMaxStyle?: CSSProperties;
+}
+
+export function BoxPlot(props: AnnotationBoxPlotProps) {
+  const {
+    medianWidth,
+    medianColor = 'black',
+    medianStyle,
+    strokeWidth,
+    strokeColor = 'black',
+    boxColor = 'none',
+    boxStyle,
+    whiskerWidth,
+    whiskerColor = 'black',
+    whiskerStyle,
+    minMaxWidth,
+    minMaxColor = 'black',
+    minMaxStyle,
+    ...position
+  } = props;
+  const { min, max, q1, median, q3, width, y, horizontal } =
+    useBoxPlotPosition(position);
+  const y1 = y - width / 2;
+  const y2 = y + width / 2;
+  const height = Math.abs(q3 - q1);
+  return (
+    <g>
+      {/* interquartile range */}
+      <rect
+        x={horizontal ? q1 : y1}
+        y={horizontal ? y1 : q1}
+        width={horizontal ? height : width}
+        height={horizontal ? width : height}
+        strokeWidth={strokeWidth}
+        stroke={strokeColor}
+        fill={boxColor}
```

Comment:
boxColor should be passed to stroke (black by default)

Reply:
Luc mentioned in the issue that the default values of the box is
No fill and black stroke

Reply:
Yes that's the point.

if we do `stroke={boxColor}`, we will have a black stroke and no fill.

Reply:
This depends on the previous suggestion which is to remove `strokeColor` and `strokeWidth` from the props.

Reply:
If we want to fill the box, we can then do `boxStyle={{fill: 'red'}}`

Reply:
Yes clear